app/src/main/java/ch/epfl/sweng/SDP/home/LeaderboardActivity.java

```diff
@@ -240,20 +244,20 @@ private void fetchPlayersFromFirebase(final String query) {
                         @Override
                         public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
                             for (DataSnapshot s : dataSnapshot.getChildren()) {
-                                if (!s.child("userId").exists() || !s.child("username").exists()
-                                        || !s.child("trophies").exists()
-                                        || !s.child("currentLeague").exists()
-                                        || s.getKey().equals("123456789")) {
-                                    continue;
+                                if (!s.getKey().equals("123456789")
+                                        && s.child("userId").getValue(String.class) != null
```

Comment:
Here we get each value twice, maybe it’s better to extract a variable for each one